packages/validate/test/ValidateMixin.test.js

```diff
@@ -1285,8 +1286,46 @@ describe('ValidateMixin', () => {
     });
   });
 
-  describe(`Asynchronous validation [to-be-implemented] ${suffixName}`, () => {
-    it('handles promises as custom validator functions [to-be-implemented]', async () => {});
+  describe('Asynchronous validation', () => {
+    it('handles promises as custom validator functions', async () => {
+
+      let asyncValidationReady;
+      const asyncValidationPromise = new Promise((resolve) => {
+        asyncValidationReady = resolve;
+      });
+
+      class DelayedCatValidator extends Validator {
+        constructor(param, config) {
+          super(param, config);
+          this.name = 'delayed-cat';
+          this.async = true;
+        }
+
+       /**
+         * @desc the function that returns a Boolean
+         * @param {string} modelValue
+         */
+        async execute(modelValue) {
+          await asyncValidationPromise;
+          return modelValue === 'cat';
+        }
+      }
+
+      const el = await fixture(html`
+        <${tag}
+          .modelValue=${'dog'}
+          .errorValidators=${[new DelayedCatValidator()]}
+        >
+        ${lightDom}
+        </${tag}>
+      `);
+      const validator = el.errorValidators[0];
+      expect(validator instanceof Validator).to.equal(true);
+      expect(el.errorState).to.equal(false);
```

Comment:
don't we need another errorState then? e.g. something like `pending` or so?

Reply:
Oh yeah, definitely. Not finished by a long shot :)
This was only to make sure the concept (having async and sync in the same array) works. I think the pending state was already described in an it below...